packages/devtools_app/test/timeline_model_test.dart

```diff
@@ -187,6 +195,47 @@ void main() {
       expect(offlineData.selectedEvent, equals(copy.selectedEvent));
       expect(offlineData.displayRefreshRate, equals(copy.displayRefreshRate));
       expect(offlineData.cpuProfileData, equals(copy.cpuProfileData));
+      expect(offlineData.timelineMode, equals(copy.timelineMode));
+      expect(identical(offlineData, copy), isFalse);
+    });
+  });
+
+  group('OfflineFullTimelineData', () {
+    test('init from parse', () {
+      OfflineFullTimelineData offlineData = OfflineFullTimelineData.parse({});
+      expect(offlineData.traceEvents, isEmpty);
+      expect(offlineData.selectedEvent, isNull);
+      expect(offlineData.cpuProfileData, isNull);
+      expect(offlineData.timelineMode, equals(TimelineMode.full));
+
+      offlineData = OfflineFullTimelineData.parse(offlineFullTimelineDataJson);
+      expect(
+        offlineData.traceEvents,
+        equals(goldenTraceEventsJson),
+      );
+      expect(offlineData.selectedEvent, isA<OfflineTimelineEvent>());
+      expect(
+        offlineData.selectedEvent.json,
+        equals({
+          TimelineEvent.eventNameKey: vsyncEvent.name,
+          TimelineEvent.eventTypeKey: vsyncEvent.type.toString(),
+          TimelineEvent.eventStartTimeKey: vsyncEvent.time.start.inMicroseconds,
+          TimelineEvent.eventDurationKey:
+              vsyncEvent.time.duration.inMicroseconds,
+        }),
+      );
+      expect(offlineData.cpuProfileData.json, equals(goldenCpuProfileDataJson));
+      expect(offlineData.timelineMode, equals(TimelineMode.full));
+    });
+
+    test('copy', () {
```

Comment:
unless you are already using the work copy elsewhere, I would suggest renaming copy to clone or shallowClone as that is the more commonly use name in Dart.